packages/venia-ui/lib/components/AddressBookPage/addEditDialog.js

```diff
@@ -0,0 +1,178 @@
+            <div className={classes.root}>
+                <div className={classes.firstname}>
+                    <Field id="firstname" label={firstNameLabel}>
+                        <TextInput field="firstname" validate={isRequired} />
+                    </Field>
+                </div>
+                <div className={classes.middlename}>
+                    <Field
+                        id="middlename"
+                        label={middleNameLabel}
+                        optional={true}
+                    >
+                        <TextInput field="middlename" />
+                    </Field>
+                </div>
+                <div className={classes.lastname}>
+                    <Field id="lastname" label={lastNameLabel}>
+                        <TextInput field="lastname" validate={isRequired} />
+                    </Field>
+                </div>
+                <div className={classes.country}>
+                    <Country field={'country_code'} validate={isRequired} />
+                </div>
+                <div className={classes.street1}>
+                    <Field id="street1" label={street1Label}>
+                        <TextInput field="street[0]" validate={isRequired} />
+                    </Field>
+                </div>
+                <div className={classes.street2}>
+                    <Field id="street2" label={street2Label} optional={true}>
+                        <TextInput field="street[1]" />
+                    </Field>
+                </div>
+                <div className={classes.city}>
+                    <Field id="city" label={cityLabel}>
+                        <TextInput field="city" validate={isRequired} />
+                    </Field>
+                </div>
+                <div className={classes.region}>
+                    <Region
+                        countryCodeField={'country_code'}
+                        fieldInput={'region[region]'}
+                        fieldSelect={'region[region_id]'}
+                        optionValueKey="id"
+                        validate={isRequired}
+                    />
+                </div>
+                <div className={classes.postcode}>
+                    <Postcode validate={isRequired} />
+                </div>
+                <div className={classes.telephone}>
+                    <Field id="telephone" label={telephoneLabel}>
+                        <TextInput field="telephone" validate={isRequired} />
+                    </Field>
+                </div>
+                <div className={classes.default_address_check}>
+                    <Checkbox
+                        disabled={isCheckboxDisabled}
+                        field="default_shipping"
+                        label={defaultAddressCheckLabel}
+                    />
+                </div>
+            </div>
+        </Dialog>
+    );
+};
+
+export default AddEditDialog;
+
+AddEditDialog.propTypes = {
+    classes: shape({}),
```

Comment:
Need to actually fill this out.

Reply:
https://github.com/magento/pwa-studio/pull/2879/commits/340c38688f60dcbbed7bc46548969a0e96d0cd92